### PR TITLE
Adding targets section for configuration

### PR DESCRIPTION
**Pull request by edwinavalos**
Will add a --target <resource> for each item in a comma separated list to the plan and apply commands

Updated the docs to show the new behavior. If left undefined there is no regression in functionality.

### CODE REVIEW

**Comment by gkuchta** [comment]
:+1:

File: plugin.go [hunk 1]
```diff
@@ -123,11 +124,15 @@ func getModules() *exec.Cmd {
...
 }
 
-func planCommand(variables map[string]string, secrets map[string]string, parallelism int) *exec.Cmd {
+func planCommand(variables map[string]string, secrets map[string]string, parallelism int, targets []string) *exec.Cmd {
 	args := []string{
 		"plan",
 		"-out=plan.tfout",
 	}
+	for _, v := range targets {
+		args = append(args, "--target")
```
Comment (jmccann) [inline plugin.go]: This line can be combined with the next to be `args = append(args, "--target", fmt.Sprintf("%s", v)) `

File: plugin.go [hunk 2]
```diff
@@ -145,10 +150,14 @@ func planCommand(variables map[string]string, secrets map[string]string, paralle
 	)
 }
 
-func applyCommand(parallelism int) *exec.Cmd {
+func applyCommand(parallelism int, targets []string) *exec.Cmd {
 	args := []string{
 		"apply",
 	}
+	for _, v := range targets {
+		args = append(args, "--target")
```
Comment (jmccann) [inline plugin.go]: This line can be combined with the next to be args = append(args, "--target", fmt.Sprintf("%s", v))

File: DOCS.md [hunk 3]
```diff
@@ -124,6 +124,26 @@ pipeline:
...
 
+## targets
+You may want to only target a specific list of resources within your terraform code. To achieve this you can specify the `targets` parameter. If left undefined all resources will be planned/applied against as the default behavior. You can define multiple targets with a comma separated list.
+
+```yaml
+pipeline:
+  terraform:
+    image: jmccann/drone-terraform:0.5
+    plan: false
+    targets: aws_security_group.generic_sg, aws_security_group.app_sg
```
Comment (tboerger) [inline DOCS.md]: Why comma separated list instead of string slice?

Comment (tboerger) [inline DOCS.md]: hint: you already use the correct flag for that

Comment (tboerger) [inline DOCS.md]: You should wrote a single string or an array in yaml format. Those are the options we also use for other drone plugins. We don't use a comma separated list anywhere

**Pull request merged**